src/Core/Extensions/SpacingExtensions.cs

```diff
@@ -0,0 +1,116 @@
+// ------------------------------------------------------------------------
+// MIT License - Copyright (c) Microsoft Corporation. All rights reserved.
+// ------------------------------------------------------------------------
+
+using System.Globalization;
+
+namespace Microsoft.FluentUI.AspNetCore.Components;
+
+/// <summary>
+/// Convert a spacing value to a CSS styleValue.
+/// </summary>
+public static class SpacingExtensions
+{
+    private static KeyValuePair<string?, (string Style, string Class)> _previousSpacingToStyle = new(key: null, value: ("", ""));
+    private static readonly char[] Separators = [' ', ';'];
+    private static readonly string[] StyleKeyWords = ["auto", "inherit", "initial", "revert", "revert-layer", "unset"];
+    private static readonly char[] InvalidCharsInClassName = ['!', '@', '#', '$', '%', '^', '&', '*', '(', ')', '+', '=', '{', '}', '[', ']', '|', '\\', ':', ';', '"', '\'', '<', '>', ',', '?', '/'];
+
+    /// <summary>
+    /// Converts a spacing value to a CSS styleValue.
+    /// The <paramref name="value"/> can be a string with one or more values separated by spaces, coma or semicolon.
+    /// You can use the `padding` or `margin` pattern defined my the W3C
+    /// or a classValue name if the value is not a valid CSS keyword like `auto`, `inherit`, `initial`, ...
+    /// </summary>
+    /// <example>
+    ///  - SpacingToStyle("auto")                  => Style = "auto"               Class = ""
+    ///  - SpacingToStyle("10px")                  => Style = "10px"               Class = ""
+    ///  - SpacingToStyle("10px 20px")             => Style = "10px 20px"          Class = ""
+    ///  - SpacingToStyle("10px 20px 30px")        => Style = "10px 20px 30px"     Class = ""
+    ///  - SpacingToStyle("mr-0")                  => Style = ""                   Class = "mr-0"
+    ///  - SpacingToStyle("my-classValue")         => Style = ""                   Class = "my-classValue"
+    ///  - SpacingToStyle("mr-0 my-classValue")    => Style = ""                   Class = "mr-0 my-classValue"
+    /// </example>
+    /// <param name="value"></param>    
+    /// <returns></returns>
+    public static (string Style, string Class) SpacingToStyle(this string? value)
```

Comment:
SpacingToStyle().Class is a bit strange 

Maybe `ConvertSpacing` so you'd get `ConvertSpacing().Class` and `ConvertSpacing().Style`?